app/src/org/commcare/engine/cases/CaseUtils.java

```diff
@@ -25,17 +25,51 @@
 import org.javarosa.core.services.storage.IStorageIterator;
 import org.javarosa.core.services.storage.IStorageUtilityIndexed;
 import org.javarosa.core.util.DAG;
+import org.javarosa.core.util.MD5;
 import org.javarosa.model.xform.XPathReference;
 
 import java.util.HashMap;
 import java.util.Vector;
 
+import static org.commcare.cases.util.CaseDBUtils.xordata;
+
 /**
  * Utilities for performing complex operations on the case database.
  *
  * Created by ctsims on 2/25/2016.
  */
 public class CaseUtils {
+
+    public static String computeCaseDbHash(SqlStorage<?> storage) {
+        long timeStart = System.currentTimeMillis();
+
+        byte[] data = new byte[MD5.length];
+        for (int i = 0; i < data.length; ++i) {
+            data[i] = 0;
+        }
+        boolean casesExist = false;
+        long count = 0;
+
+        for (SqlStorageIterator i = storage.iterate(false, new String[]{Case.INDEX_CASE_ID}); i.hasMore(); ) {
+            byte[] current = MD5.hash(i.peekIncludedMetadata(Case.INDEX_CASE_ID).getBytes());
+            data = xordata(data, current);
+            casesExist = true;
+            count++;
+            i.nextID();
+        }
+
+        //In the base case (with no cases), the case hash is empty
+        if (!casesExist) {
+            return "";
+        }
+
+        long timeEnd = System.currentTimeMillis();
+        Logger.log(LogTypes.TYPE_MAINTENANCE, String.format(
```

Comment:
Do we want to continue logging time here ? (Just flagging in case it was just meant for comparing the 2 methods)

Reply:
Good catch, thanks for bringing it up. I left it in because we also do this measurement in the Case Purge Process:
https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/engine/cases/CaseUtils.java#L113

and this actually runs less frequently than the purge does, but definitely good for us to keep track of.